vdr/ownership-checker.go

```diff
@@ -0,0 +1,25 @@
+package vdr
+
+import (
+	"github.com/nuts-foundation/go-did/did"
+
+	"github.com/nuts-foundation/nuts-node/crypto"
+	"github.com/nuts-foundation/nuts-node/vdr/types"
+)
+
+type OwnershipChecker struct {
+	KeyResolver KeyResolver
+	KeyChecker  crypto.PrivateKeyChecker
+}
+
+func (c OwnershipChecker) OwnedByThisNode(id did.DID) (error) {
```

Comment:
Just let VDR implement an OwnershipChecker interface?

And although it's a high level function,  I think the name is extremely high level (`node ownership`), possibly too high level for `VDR`. We planned to have `didman` for very high level stuff imo.

`Assertable` (Wout's suggestion) sounds maybe a bit (too) technical to me? Maybe `CanManage`/`CanUpdate`?

Reply:
Imo this is a combined functionality over the keystore and the document store. So neither one of those should implement it. It is meant to be an Authorization cheker if a user should be able to update a DID Document or can use a key to sign a token or create a VC etc.